### DISCUSSION TITLE
Storefronts V2

### DISCUSSION THREAD

**Discussion by ericingram** [swellstores/community#574] [Announcements]
Hey everyone 👋  We recently announced details of a new storefront editor, designed to help merchants customize and launch an online store in minutes. Based on our app framework, storefront apps support themes and extensions to build any kind of shopping experience. Customize our default theme, build your own, or even upload a Shopify 2.0 theme. Early access opens November 2024.

https://www.swell.is/features/storefront-editor

**Here are some more details about Storefronts V2**

- A storefront app can introduce new content models, serverless functions, and other Swell App configurations along with a hosted frontend.

- Our first storefront app is an open-source project called Proxima, built with [Astro](https://astro.build/), and works with our own flavor of Liquid templates as well as Shopify 2.0 themes you can upload directly.

- A store can have one live storefront and a number of development versions.

- The new Swell CLI allows developers to build and test storefront apps and themes locally.

- Storefront apps can optionally support multiple themes and make use of our new visual theme editor.

- The theme editor should feel familiar to merchants who have used Shopify, with flexible sections, blocks, and inline editing.

**Future considerations**

- App blocks, allowing 3rd party apps to enhance the functionality of a storefront theme.

- App Store for developers to distribute both open-source or paid apps and themes.

Join our [Discord channel](https://discord.com/channels/1001781312899796994/1285374808116297728) and let us know what features you're curious about or hope to see.

**Comment by ezswan** [comment]
So, the builder is built with Astro, but the outputted websites are liquid? Liquid is not ideal. Now a website builder that builds astro websites, well that would be a game changer.

**Reply by ericingram** [reply]
It's possible to do both in this framework. Our first storefront app will support Liquid themes, but we expect more apps will be released and open-sourced using different frameworks, with and without themes.

**Reply by mcconnon12** [reply]
When will this be rolling out to everyone?

**Reply by ericingram** [reply]
Soon

**Reply by mcconnon12** [reply]
Do you have an ETA? I don't want to be building something if a solution is coming down the pipeline. Thanks.

**Reply by ericingram** [reply]
Right now we're aiming for the week of Feb 17 to start sending invites

**Reply by mcconnon12** [reply]
Thanks for that Eric... am I correct in assuming it will have a customer portal for customers to manage their accounts?

Anyway I can get on that early invite list 🙂?

**Reply by ericingram** [reply]
Of course! If you already added your email at https://www.swell.is/features/storefront-editor then we'll be reaching out soon

**Reply by mcconnon12** [reply]
Hey Eric, I see a notification under storefront about feedback for the V2 beta but it doesn't look like I have access. I haven't received any email notifications or anything so I'm guessing it just hasn't rolled out to me yet?

**Reply by ericingram** [reply]
It hasn't rolled out yet, but we're almost there :)

**Reply by mcconnon12** [reply]
Thanks!

**Comment by mcconnon12** [comment]
Thanks... And the storefront will include a customer portal for customers as well?

**Reply by ericingram** [reply]
Yes that's right
